internal/clients/interface.go

```diff
@@ -109,6 +109,17 @@ type Azure interface {
 	// EnsureResourceGroup makes sure that group with give name exists in a location
 	EnsureResourceGroup(ctx context.Context, name string, location string) (*string, error)
 
+	// BeginCreateVM starts process of creating Azure virtual machine.
+	// Most of the parameters are constant for now.
+	// location - to deploy into
+	// imageName - the imageID will be inferred as /subscriptions/{subscriptionID}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/images/{imageName}
+	// returns a resumeToken
+	BeginCreateVM(ctx context.Context, location string, resourceGroupName string, imageName string, pubkey *models.Pubkey, instanceType InstanceTypeName, vmName string) (string, error)
+
+	// WaitForVM starts polling of status of Azure virtual machine creation and wait until it's done.
+	// needs resumeToken from BeginCreateVM operation.
+	WaitForVM(ctx context.Context, resumeToken string) (*string, error)
+
 	// CreateVM creates Azure virtual machine.
```

Comment:
I understand why you want to split begin and wait, but do you really need to expose both via the interface? I mean, these both could be unexported functions and callers would continue using `CreateVM` as usual?

Reply:
Sure, we can start with that and if we want to expose this to multiple steps, we can get back to this solution :+1:

Reply:
Thanks I would appreciate the interface to be simple, because something tells me we might be removing the waiting pretty soon. :-D